test/test_io.py

```diff
@@ -76,39 +77,40 @@ def test_read_timestamps(self):
             # so we use it as a baseline
             container = av.open(f_name)
             stream = container.streams[0]
-            pts_step = int(round(float(1 / (stream.average_rate * stream.time_base))))
             num_frames = int(round(float(stream.average_rate * stream.time_base * stream.duration)))
-            expected_pts = [i * pts_step for i in range(num_frames)]
+            # pts is a global fraction of a second, we expect
+            # to see num_frames frames whose global pts is 1/fps
+            expected_pts = [Fraction(i, stream.average_rate) for i in range(num_frames)]
 
             self.assertEqual(pts, expected_pts)
 
     def test_read_partial_video(self):
         with temp_video(10, 300, 300, 5, lossless=True) as (f_name, data):
-            pts, _ = io.read_video_timestamps(f_name)
+            pts, fps = io.read_video_timestamps(f_name)
             for start in range(5):
                 for l in range(1, 4):
                     lv, _, _ = io.read_video(f_name, pts[start], pts[start + l - 1])
                     s_data = data[start:(start + l)]
                     self.assertEqual(len(lv), l)
                     self.assertTrue(s_data.equal(lv))
 
-            lv, _, _ = io.read_video(f_name, pts[4] + 1, pts[7])
+            lv, _, _ = io.read_video(f_name, pts[4] + 1 / (fps + 1), pts[7])
```

Comment:
Can you explain the need of the `+1` in `(fps + 1)`?

Reply:
My understanding was that the original test was here to make sure that if we try to decode from a pts that doesn't exist (`pts[4]+1`) that we return the closest possible frame to that pts - then we check and get 4 frames (from `pts[4]` to `pts[7]`).

In the same way, `1/(fps+1)` is a non-existing pts that is the closest to the existing (`pts[4]`).